dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/SubmissionFormConverter.java

```diff
@@ -106,23 +107,26 @@ private SubmissionFormFieldRest getField(DCInput dcinput, String formName) {
         inputRest.setRegex(dcinput.getRegex());
 
         if (!StringUtils.equalsIgnoreCase(dcinput.getInputType(), "qualdrop_value")) {
-            // value-pair and vocabulary are a special kind of authorities
-            String inputType = dcinput.getInputType();
-
-            SelectableMetadata selMd = new SelectableMetadata();
-            if (authorityUtils.isChoice(dcinput.getSchema(), dcinput.getElement(), dcinput.getQualifier())) {
-                inputRest.setType(
-                        getPresentation(dcinput.getSchema(), dcinput.getElement(), dcinput.getQualifier(), inputType));
-                selMd.setAuthority(getAuthorityName(dcinput.getSchema(), dcinput.getElement(), dcinput.getQualifier(),
-                        dcinput.getPairsType(), dcinput.getVocabulary()));
-                selMd.setClosed(
-                        authorityUtils.isClosed(dcinput.getSchema(), dcinput.getElement(), dcinput.getQualifier()));
-            } else {
-                inputRest.setType(inputType);
+            if (dcinput.isMetadataField()) {
```

Comment:
It's unclear to me why this section only applies when `isMetadataField` is true.  Could we add a comment here as to why the new `if` is necessary?  Also, should this `if` and the one above be combined into a single `if` (and the comment/explanation added there)?

Reply:
some inline comments have been included to clarify the reason for the current `if` statements